test/Transforms/memory_bank_invalid.mlir

```diff
@@ -0,0 +1,14 @@
+// RUN: circt-opt %s -split-input-file -memory-banking="banking-factor=0" -verify-diagnostics
```

Comment:
Nit: keep the naming consistent: `memory_banking_invalid.mlir` or cahnge the other to `memory_bank.mlir`